File: riveredge-frontend/src/tree-stem/services/userTask.ts
/**
 * 用户任务管理服务
 * 
 * 提供用户任务的查询、处理等功能。
 * 注意：所有 API 自动获取当前用户的任务。
 */

import { apiRequest } from './api';

export interface UserTask {
  uuid: string;
  tenant_id: number;
  process_uuid?: string;
  title: string;
  content?: string;
  data?: Record<string, any>;
  submitter_id: number;
  current_approver_id?: number;
  status: string;
  current_node?: string;
  submitted_at: string;
  completed_at?: string;
  created_at: string;
  updated_at: string;
}

export interface UserTaskListResponse {
  items: UserTask[];
  total: number;
  page: number;
  page_size: number;
}

export interface UserTaskStats {
  total: number;
  pending: number;
  approved: number;
  rejected: number;
  submitted: number;
}

export interface TaskActionRequest {
  action: 'approve' | 'reject';
  comment?: string;
}

/**
 * 获取当前用户任务列表
 */
export async function getUserTasks(params?: {
  page?: number;
  page_size?: number;
  status?: string;
  task_type?: 'pending' | 'submitted';
}): Promise<UserTaskListResponse> {
  return apiRequest<UserTaskListResponse>('/personal/user-tasks', {
    params,
  });
}

/**
 * 获取当前用户任务详情
 */
export async function getUserTask(taskUuid: string): Promise<UserTask> {
  return apiRequest<UserTask>(`/personal/user-tasks/${taskUuid}`);
}

/**
 * 获取当前用户任务统计
 */
export async function getUserTaskStats(): Promise<UserTaskStats> {
  return apiRequest<UserTaskStats>('/personal/user-tasks/stats');
}

/**
 * 处理用户任务（审批或拒绝）
 */
export async function processUserTask(
  taskUuid: string,
  data: TaskActionRequest
): Promise<UserTask> {
  return apiRequest<UserTask>(`/personal/user-tasks/${taskUuid}/process`, {
    method: 'POST',
    data,
  });
}

File: riveredge-backend/src/core/schemas/user_task.py
"""
用户任务 Schema 模块

定义用户任务相关的 Pydantic Schema，用于数据验证和序列化。
复用 ApprovalInstance 模型，但提供用户视角的 Schema。
"""

from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, Dict, Any, List
from datetime import datetime
from uuid import UUID


class UserTaskResponse(BaseModel):
    """用户任务响应 Schema（复用 ApprovalInstance）"""
    uuid: UUID = Field(..., description="任务UUID")
    tenant_id: int = Field(..., description="组织ID")
    process_uuid: Optional[UUID] = Field(None, description="关联审批流程UUID")
    title: str = Field(..., description="任务标题")
    content: Optional[str] = Field(None, description="任务内容")
    data: Optional[Dict[str, Any]] = Field(None, description="审批数据（JSON格式）")
    submitter_id: int = Field(..., description="提交人ID")
    current_approver_id: Optional[int] = Field(None, description="当前审批人ID（当前用户）")
    status: str = Field(..., description="任务状态（pending、approved、rejected、cancelled）")
    current_node: Optional[str] = Field(None, description="当前节点")
    submitted_at: Optional[datetime] = Field(None, description="提交时间")
    completed_at: Optional[datetime] = Field(None, description="完成时间")
    created_at: Optional[datetime] = Field(None, description="创建时间")
    updated_at: Optional[datetime] = Field(None, description="更新时间")
    
    model_config = ConfigDict(from_attributes=True)


class UserTaskListResponse(BaseModel):
    """用户任务列表响应 Schema"""
    items: List[UserTaskResponse] = Field(..., description="任务列表")
    total: int = Field(..., description="总数量")
    page: int = Field(..., description="当前页码")
    page_size: int = Field(..., description="每页数量")


class UserTaskStatsResponse(BaseModel):
    """用户任务统计响应 Schema"""
    total: int = Field(..., description="总任务数")
    pending: int = Field(..., description="待处理任务数")
    approved: int = Field(..., description="已通过任务数")
    rejected: int = Field(..., description="已拒绝任务数")
    submitted: int = Field(..., description="我提交的任务数")


class UserTaskActionRequest(BaseModel):
    """用户任务操作请求 Schema"""
    action: str = Field(..., description="操作类型（approve、reject）")
    comment: Optional[str] = Field(None, description="审批意见")

